src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -127,9 +130,18 @@ internal class DiagnosticComputer
 
             if (logPerformanceInfo && _performanceTracker != null)
             {
-                // +1 to include project itself
-                var unitCount = documentAnalysisScope != null ? 1 : _project.DocumentIds.Count + 1;
-                _performanceTracker.AddSnapshot(analysisResult.AnalyzerTelemetryInfo.ToAnalyzerPerformanceInfo(_analyzerInfoCache), unitCount);
+                // Only log telemetry snapshot is we have an active telemetry session,
+                // i.e. user has not opted out of reporting telemetry.
+                var telemetryService = _hostWorkspaceServices.GetRequiredService<IWorkspaceTelemetryService>();
+                if (telemetryService.HasActiveSession)
```

Comment:
❔ Should this be using `IsOptedIn` to match the other location?

Reply:
That property is exposed off the session, not the service. The check here basically checks if there is a non-null underlying CurrentSession and CurrentSession.IsOptedIn is true.